### ISSUE TITLE
Python: module-doc-string not recognized if module starts with a comment

### ISSUE THREAD

**Issue by thom-sch** [doxygen/doxygen#9188]
If a python module starts with comment lines (lines beginning with an "#") subsequent module doc string is not recognized.
An exception is a shebang comment as first line.
This behavior is independent from configuration option PYTHON_DOCSTRING

**Expected behavior**
Python allows a coding string (also preceeded by "#") as first or second line; thus Dexygen shold at least recognize this as valid comment before the module doc string, too.
Additionally it may be useful to recognize the module doc string regardless of the number of comment lines ("#" or "##") preceeding it.

**To Reproduce**
s. attached ZIP with default configuration option file:
[Test17_python__doc__string.zip](https://github.com/doxygen/doxygen/files/8196608/Test17_python__doc__string.zip)

**Version**
- Win10/64
- configuration: INPUT                  = .
- doxgen version: 1.9.3 (c0b9eafbfb53286ce31e75e2b6c976ee4d345473)

Reagards from Meerbusch
Thomas

**Comment by albert-github** [comment]
Looks like that the shebang is explicitly ignored when it is on the first line of the code (from pyscanner.l):
```
    {SCRIPTCOMMENT}   { // Unix type script comment
                        if (yyextra->yyLineNr != 1) REJECT;
                      }
```
For normal `#` comments the following rule applies:
```
    {POUNDCOMMENT}    { // normal comment
                        yyextra->packageCommentAllowed = FALSE;
                      }
```
but probably a more sophisticated detection of the line has to take place (i.e. as long as we only have empty or single (!) `#` lines don't set the `yyextra->packageCommentAllowed = FALSE;`.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9193

**Comment by thom-sch** [comment]
sorry for late answer - I had to interrupt my tentative steps on Python due to other work...

Now I've tested your patch and it works fine for me.
Seem's that Doxygen now simply accepts any number of comment lines preceeding the docstring?

Regards from Meerbusch
Thomas

**Comment by albert-github** [comment]
Looks like the problem has been solved by an even simpler fix see the Issue #9672 / PR #9673

**Comment by thom-sch** [comment]
Hello Albert,

thanks for the info.
I've just done a short test - looks well on my example and on my real life project, too.

Feel free to close this issue.

Regards from Meerbusch
Thomas
